webpack.config.client.js

```diff
@@ -73,21 +75,79 @@ module.exports = ({
       ],
       // specify min/max file sizes for each JS chunk for optimal performance
       splitChunks: {
-        chunks: 'initial',
+        chunks: 'all',
         automaticNameDelimiter: '-',
-        minSize: 184320, // 180kb
         maxSize: 245760, // 240kb
         cacheGroups: {
-          common: {
-            name: false,
-            minChunks: 2,
+          default: false,
+          vendors: false,
+          framework: {
+            name: 'framework',
             chunks: 'all',
+            // This regex ignores nested copies of framework libraries so they're bundled with their issuer.
+            test: new RegExp(
+              `(?<!node_modules.*)[\\\\/]node_modules[\\\\/](${FRAMEWORK_BUNDLES.join(
+                `|`,
+              )})[\\\\/]`,
+            ),
+            priority: 40,
+            // Don't let webpack eliminate this chunk (prevents this chunk from becoming a part of the commons chunk)
+            enforce: true,
+          },
+          commons: {
+            name: 'commons',
+            // if a chunk is used on all pages we put it in commons
+            minChunks: TOTAL_PAGES,
+            priority: 20,
           },
-          vendor: {
-            test: /[\\/]node_modules[\\/]/,
-            name: 'vendor',
+          lib: {
+            // if a module is bigger than 160kb from node_modules we make a separate chunk for it
+            test(module) {
+              return (
+                module.size() > 160000 &&
+                /node_modules[/\\]/.test(module.identifier())
+              );
+            },
+            name(module) {
+              const rawRequest =
+                module.rawRequest &&
+                module.rawRequest.replace(/^@(\w+)[/\\]/, '$1-');
+              if (rawRequest) return `${rawRequest}-lib`;
+
+              const identifier = module.identifier();
+              const trimmedIdentifier = /(?:^|[/\\])node_modules[/\\](.*)/.exec(
+                identifier,
+              );
+              const processedIdentifier =
+                trimmedIdentifier &&
+                trimmedIdentifier[1].replace(/^@(\w+)[/\\]/, '$1-');
+
+              return `${processedIdentifier || identifier}-lib`;
+            },
+            priority: 30,
+            minChunks: 1,
+            reuseExistingChunk: true,
+          },
+          shared: {
+            name(module, chunks) {
+              const cryptoName = crypto
+                .createHash('sha1')
+                .update(
+                  chunks.reduce((acc, chunk) => {
+                    return acc + chunk.name;
+                  }, ''),
+                )
+                .digest('base64')
+                .replace(/\//g, '');
+
+              return `shared-${cryptoName}`;
+            },
+            priority: 10,
+            minChunks: 2,
+            reuseExistingChunk: true,
           },
         },
+        maxInitialRequests: 20,
```

Comment:
what does this do?

Reply:
From the Next.js PR linked in the issue

> Note that SplitChunksPlugin has configuration options to prevent runaway granularity, such as a situation where every module ends up in its own chunk, and the user has to download 100 chunks just to render a route. In particular, in the config above, I'm using the maxInitialRequests option to prevent the number of bundles from skyrocketing.